sds.c

```diff
@@ -1018,7 +1018,14 @@ sds *sdssplitargs(const char *line, int *argc) {
                 if (*p) p++;
             }
             /* add the token to the vector */
-            vector = s_realloc(vector,((*argc)+1)*sizeof(char*));
+            char **data = s_realloc(vector,((*argc)+1)*sizeof(char*));
+            if (data == NULL) {
+                /* Return something not NULL. */
+                return s_malloc(sizeof(void*));
```

Comment:
What part of the code would re-try an allocation in order to return this memory uninitialized (if it succeeds)?
That looks like a serious bug waiting to happen from the user-perspective. How would the user even know that this memory does not hold any valid data?